.teamcity/Gradle_Check/projects/FunctionalTestProject.kt

```diff
@@ -22,15 +21,8 @@ class FunctionalTestProject(model: CIBuildModel, testConfig: TestCoverage, stage
             return@forEach
         }
 
-        if (bucket is SubprojectSplit) {
-            // https://github.com/gradle/gradle-private/issues/2693
-            if (testConfig.testType.supportTestSplit) {
-                buildType(FunctionalTest(model, testConfig, bucket.getSubprojectNames(), stage, bucket.name, bucket.extraParameters()))
-            } else if (bucket.number == 1) {
-                buildType(FunctionalTest(model, testConfig, bucket.getSubprojectNames(), stage, bucket.name))
-            }
-        } else {
-            buildType(FunctionalTest(model, testConfig, bucket.getSubprojectNames(), stage, bucket.name, bucket.extraParameters()))
+        for (buildTypeBucket in bucket.forTestType(testConfig.testType)) {
+            buildType(FunctionalTest(model, testConfig, buildTypeBucket.getSubprojectNames(), stage, buildTypeBucket.name, buildTypeBucket.extraParameters()))
```

Comment:
Note that when we're not splitting the project, we should not add `-testSplit=1/3` parameter.

Reply:
And we don't: if we are not splitting the project, then `bucket.forTestType` will return a buildTypeBucket with `extraParameters == ''`.

Reply:
OK, yes you're right, it returns `listOf(subproject)`, not `listOf(this)`.